src/types/GetTokenInfo.ts

```diff
@@ -0,0 +1,7 @@
+import { Logger, Token } from '.';
+
+type GetTokenInfo<T> = (tokenInfoUrl: string, accessToken: string, logger?: Logger) => Promise<Token<T>>;
```

Comment:
We should also use this type for our implementation of `getTokenInfo`: https://github.com/zalando-incubator/authmosphere/blob/master/src/oauth-tooling.ts#L156

Then that would look like:

```ts
const getTokenInfo: GetTokenInfo = (tokenInfoUrl: string, accessToken: string, logger?: Logger) => {
  // ...
```

To make this possible, the type has to provide a default generic type:

```ts
type GetTokenInfo<T = {}> = (tokenInfoUrl: string, accessToken: string, logger?: Logger) => Promise<Token<T>>;
```

Reply:
Ah, got it! I tried to, but couldn’t  make it because I was missing `<T = {}>`.